src/OpenLoco/Entities/Misc.cpp

```diff
@@ -126,18 +126,86 @@ namespace OpenLoco
 
     static loco_global<int32_t, 0x112C876> _currentFontSpriteBase;
 
+    // 0x004FADD0
+    constexpr Map::Pos2 _wiggleAmounts[] = {
+        { 1, -1 },
+        { 1, 1 },
+        { -1, 1 },
+        { -1, -1 },
+    };
+
+    // clang-format off
+    // 0x004FAD21
+    constexpr int8_t _wiggleZAmounts[] = {
+        3, 3, 3, 3, 3, 3, 3, 3, 3, 3,
+        2, 1, 0, 0, 0, 1, 0, 0, 0, 1,
+        0, 0, 0, 1, 0, 0, 0, 1, 0, 0,
+        0, 1, 0, 0, 0, 1, 0, 0, 0, 1,
+        0, 0, 0, 1, 0, 0, 0, 1, 0, 0,
+        0, 1, 0, 0, 0, 1, 0, 0, 0, 1,
+        0, 0, 0, 1, 0, 0, 0, 1, 0, 0,
+        0, 1, 0, 0, 0, 1, 0, 0, 0, 1,
+        0, 0, 0, 1, 0, 0, 0, 1, 0, 0,
+        0, 1, 0, 0, 0, 1, 0, 0, 0, 1,
+        0, 0, 0, 1, 0, 0, 0, 1, 0, 0,
+        0, 1, 0, 0, 0, 1, 0, 0, 0, 1,
+        0, 0, 0, 1, 0, 0, 0, 1, 0, 0,
+        0, 1, 0, 0, 0, 1, 0, 0, 0, 1,
+        0, 0, 0, 1, 0, 0, 0, 1, 0, 0,
+        0, 1, 0, 0, 0, 1, 0, 0, 0, 1,
+    };
+    // clang-format on
+
     // 0x0044063E
     void MoneyEffect::update()
     {
-        registers regs;
-        regs.esi = X86Pointer(this);
         if (getSubType() == MiscEntityType::windowCurrency)
         {
-            call(0x0044063E, regs);
+            invalidateSprite();
+            if (wiggle == 0)
+            {
+                wiggle = 21;
+            }
+            else
+            {
+                wiggle--;
+            }
+
+            if (frame >= 160)
```

Comment:
it only just clicked in my head that this 160 is the number of elements in `_wiggleZAmounts` - perhaps you could make this a constant like `constexpr int8_t kWiggleAmountCount = sizeof(_wiggleZAmounts) / sizeof(std::remove_all_extents<decltype(_wiggleZAmounts)>::type);`

or if you don't care about being so generic, 
constexpr int8_t kWiggleAmountCount = sizeof(_wiggleZAmounts) / sizeof(_wiggleZAmounts[0]);

Reply:
I'm sort of doing it in the reverse. So could just do kLifetime = 160 then make the array fixed to that size

Reply:
Ah yeah true, that works for me

Reply:
Okay ive added two constants for it.

Reply:
Looks good to me